src/ConfigDiscovery/AbstractDiscovery.php

```diff
@@ -33,10 +33,8 @@ abstract class AbstractDiscovery implements DiscoveryInterface
      *
      * Optionally specify project directory; $configFile will be relative to
      * this value.
-     *
-     * @param string $projectDirectory
      */
-    public function __construct($projectDirectory = '')
+    public function __construct(string $projectDirectory = '')
```

Comment:
This change introduce compatibility break, because changes public interface

Reply:
I see your point.

IIRC I discussed this with @boesing in chat. The `@param` type hint states parameter is of type string. If implementation is breaking, it is due to a misuse.

Is that correct, @boesing?

Reply:
Sadly, `__construct` has some special behavior if it comes to arguments. So this is actually no BC break until the type-hint matches what we do state in the `@var` annotation.

(sidenote: imho `laminas-component-installer` is not a library and thus, semver does not really make sense here at all. But actually, we do not differ between libraries and components like this (composer plugin))

Reply:
fun-fact: this constructor argument is only added for unit tests so we can pass virtual filesystem directory